### PR TITLE
New geometry - dotplot

### PR DESCRIPTION
**Pull request by ASmirnov-HORIS**

### CODE REVIEW

File: plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/geom/DotplotGeom.kt [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+        p: DataPointAesthetics,
+        dotId: Int,
+        ctx: GeomContext,
+        geomHelper: GeomHelper,
+        binWidthPx: Double
+    ) {
+        if (!isFinite(p.x()) || !isFinite(p.stacksize()))
+            return
+
+        val dotRadius = dotsize * binWidthPx
```
Comment (alshan) [inline plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/geom/DotplotGeom.kt:102]: Did you mean `dotDiameter` ?

**Review by alshan — CHANGES_REQUESTED** [review]

File: plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/geom/DotplotGeom.kt (hunk 1 above)
Comment (IKupriyanov-HORIS) [inline plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/geom/DotplotGeom.kt:102]: Also `dotsize` should be `dotSize`

File: plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/GeomMeta.kt [hunk 2]
```diff
@@ -101,6 +101,17 @@ object GeomMeta {
...
             )
 
+            GeomKind.DOTPLOT -> listOf(
+                Aes.X,
+                Aes.BINWIDTH,
+                Aes.STACKSIZE,
+                Aes.COLOR,
+                Aes.FILL,
+                Aes.ALPHA,
+                Aes.WIDTH,
```
[the remark this answers is not in the document]
Comment (ASmirnov-HORIS) [reply plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/GeomMeta.kt:111]: It affects in `computeLayerDryRunXYRangesAfterSizeExpand()`. Without `Aes.WIDTH` domain for `X` is not expaned (or I should use `BINWIDTH` aes for expansion), compare this:

image

[the remark this answers is not in the document]
Comment (alshan) [reply plot-base-portable/src/commonMain/kotlin/jetbrains/datalore/plot/base/GeomMeta.kt:111]: In this case we need to add handling of `BINWIDTH` to the `computeLayerDryRunXYRangesAfterSizeExpand()`.

**Pull request merged**